From e24d09457596dac8f56895e84ddce17c31b5ff5e Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Thu, 28 Sep 2023 20:04:12 +0200
Subject: [PATCH 514/841] feat(stm32mp21): add SoC part numbers

Add STM32MP21xx SoCs part numbers and the code to display it at boot.
STM32MP21x is always single core, directly return true in function
stm32mp_is_single_core().
As for STM32MP25, STM32MP21xC and STM32MP21xF can support
authenticated boot.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 1fcd99e38fcd313ac683f9414aee081b41312ba3 feat(stm32mp21): add SoC part numbers]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: I586b584a3dab61ea9507a2c798a02ebdc9de0052
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/356305
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h     | 19 ++++++++++
 plat/st/stm32mp2/stm32mp2_private.c | 56 ++++++++++++++++++++++++++++-
 2 files changed, 74 insertions(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index e724f9aec..95118d61e 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -50,6 +50,24 @@
 /*******************************************************************************
  * CHIP ID
  ******************************************************************************/
+#if STM32MP21
+#define STM32MP2_CHIP_ID	U(0x503)
+
+#define STM32MP211A_PART_NB	U(0x40073E7D)
+#define STM32MP211C_PART_NB	U(0x0007307D)
+#define STM32MP211D_PART_NB	U(0xC0073E7D)
+#define STM32MP211F_PART_NB	U(0x8007307D)
+#define STM32MP213A_PART_NB	U(0x40073E1D)
+#define STM32MP213C_PART_NB	U(0x0007301D)
+#define STM32MP213D_PART_NB	U(0xC0073E1D)
+#define STM32MP213F_PART_NB	U(0x8007301D)
+#define STM32MP215A_PART_NB	U(0x40033E0D)
+#define STM32MP215C_PART_NB	U(0x0003300D)
+#define STM32MP215D_PART_NB	U(0xC0033E0D)
+#define STM32MP215F_PART_NB	U(0x8003300D)
+#endif /* STM32MP21 */
+
+#if STM32MP25
 #define STM32MP2_CHIP_ID	U(0x505)
 
 #define STM32MP251A_PART_NB	U(0x400B3E65)
@@ -68,6 +86,7 @@
 #define STM32MP257C_PART_NB	U(0x00002000)
 #define STM32MP257D_PART_NB	U(0xC0002E00)
 #define STM32MP257F_PART_NB	U(0x80002000)
+#endif /* STM32MP25 */
 
 #if STM32MP21
 #define STM32MP2_REV_A		U(0x0000)
diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index 88e6de31d..53932682c 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, STMicroelectronics - All Rights Reserved
+ * Copyright (c) 2023-2024, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -236,6 +236,45 @@ void stm32mp_get_soc_name(char name[STM32_SOC_NAME_SIZE])
 
 	/* MPUs Part Numbers */
 	switch (get_part_number()) {
+#if STM32MP21
+	case STM32MP211A_PART_NB:
+		cpu_s = "211A";
+		break;
+	case STM32MP211C_PART_NB:
+		cpu_s = "211C";
+		break;
+	case STM32MP211D_PART_NB:
+		cpu_s = "211D";
+		break;
+	case STM32MP211F_PART_NB:
+		cpu_s = "211F";
+		break;
+	case STM32MP213A_PART_NB:
+		cpu_s = "213A";
+		break;
+	case STM32MP213C_PART_NB:
+		cpu_s = "213C";
+		break;
+	case STM32MP213D_PART_NB:
+		cpu_s = "213D";
+		break;
+	case STM32MP213F_PART_NB:
+		cpu_s = "213F";
+		break;
+	case STM32MP215A_PART_NB:
+		cpu_s = "215A";
+		break;
+	case STM32MP215C_PART_NB:
+		cpu_s = "215C";
+		break;
+	case STM32MP215D_PART_NB:
+		cpu_s = "215D";
+		break;
+	case STM32MP215F_PART_NB:
+		cpu_s = "215F";
+		break;
+#endif /* STM32MP21 */
+#if STM32MP25
 	case STM32MP251A_PART_NB:
 		cpu_s = "251A";
 		break;
@@ -284,6 +323,7 @@ void stm32mp_get_soc_name(char name[STM32_SOC_NAME_SIZE])
 	case STM32MP257F_PART_NB:
 		cpu_s = "257F";
 		break;
+#endif /* STM32MP25 */
 	default:
 		cpu_s = "????";
 		break;
@@ -349,6 +389,9 @@ void stm32mp_print_boardinfo(void)
 /* Return true when SoC provides a single Cortex-A35 core, and false otherwise */
 bool stm32mp_is_single_core(void)
 {
+#if STM32MP21
+	return true;
+#else /* STM32MP21 */
 	bool single_core = false;
 
 	switch (get_part_number()) {
@@ -363,6 +406,7 @@ bool stm32mp_is_single_core(void)
 	}
 
 	return single_core;
+#endif /* STM32MP21 */
 }
 
 /* Return true when device is in closed state */
@@ -401,6 +445,15 @@ bool stm32mp_is_auth_supported(void)
 	bool supported = false;
 
 	switch (get_part_number()) {
+#if STM32MP21
+	case STM32MP211C_PART_NB:
+	case STM32MP211F_PART_NB:
+	case STM32MP213C_PART_NB:
+	case STM32MP213F_PART_NB:
+	case STM32MP215C_PART_NB:
+	case STM32MP215F_PART_NB:
+#endif /* STM32MP21 */
+#if STM32MP25
 	case STM32MP251C_PART_NB:
 	case STM32MP251F_PART_NB:
 	case STM32MP253C_PART_NB:
@@ -409,6 +462,7 @@ bool stm32mp_is_auth_supported(void)
 	case STM32MP255F_PART_NB:
 	case STM32MP257C_PART_NB:
 	case STM32MP257F_PART_NB:
+#endif /* STM32MP25 */
 		supported = true;
 		break;
 	default:
-- 
2.47.2

